solutions/USACO jan16-subseq-summing-to-sevens.mdx

```diff
@@ -1,44 +1,123 @@
 ---
 id: usaco-595
-source: USACO Bronze 2016 January
+source: USACO Silver 2016 January
 title: Subsequences Summing to Sevens
-author: thetazero
+author: Kevin Sheng, thetazero
 ---
 
 [Official Analysis](http://www.usaco.org/current/data/sol_div7_silver_jan16.html)
 
 <LanguageSection>
+
 <CPPSection>
 
-## Implementation
+## C++ Implementation
 
 ```cpp
-#include <bits/stdc++.h>
+#include <iostream>
+#include <vector>
+#include <algorithm>
 
 using namespace std;
 
-int main(){
+const int MOD = 7;
+
+int main() {
 	freopen("div7.in", "r", stdin);
 	freopen("div7.out", "w", stdout);
-	int n;
-	cin >> n;
-	vector<int> pref(n+1);
-	int ans=0;
-	vector<int> first(7, -1);
-	first[0]=0;
-	for(int i=1;i<=n;i++){
-		int x;
-		cin >> x;
-		pref[i]=(pref[i-1]+x)%7;
-		if(first[pref[i]]==-1){
-			first[pref[i]]=i;
-		}else{
-			ans=max(ans, i-first[pref[i]]);
+	int cow_num;
+	cin >> cow_num;
+
+	int longest_photo = 0;
+	vector<int> first(MOD, -1);
+	first[0] = 0;
+
+    int curr_remainder = 0;
+	for (int i = 1; i <= cow_num; i++) {
+		int cow;
+		cin >> cow;
+		
+        curr_remainder = (curr_remainder + cow) % MOD;
+		if (first[curr_remainder] == -1){
+			first[curr_remainder] = i;
+		} else {
+			longest_photo = max(longest_photo, i - first[curr_remainder]);
 		}
 	}
-	cout << ans << endl;
+	cout << longest_photo << endl;
 }
 ```
 
 </CPPSection>
+
+<JavaSection>
+
+## Java Implementation
+
+```java
```

Comment:
the official solution is in java, so i think this is unnecessary

Reply:
not really tbh, and even then what's the harm?

Reply:
I think the idea is basically the same, and it's preferred to not duplicate solutions... what do other people think?

>It is **not** necessary to add an alternative implementation in the same language as the official analysis (unless the alternative implementation takes a different approach or is written better / differently from the official implementation).